README.md

```diff
@@ -1,13 +1,238 @@
 ![Kin Token](kin_android.png)
-# move-kin-android
-This repository contains modules that connect the ecosystem apps together.
+# Overview
+This repository contains a module which connects all the ecosystem apps together.
+
 
 ## Ecosystem Apps Discovery Module
-Documentation coming up very soon!
-In the meantime take a look at our `Sample` apps.
+This Module provides the ability to users to explore other apps in the Kin Ecosystem and get familiar with the different experiences users can enjoy using Kin
+It also allows users to send/receive kin to/from different ecosystem applications.
+
+
+## Installation
+To include the library in your project add jitpack to your *build.gradle* project file.
+
+```gradle
+allprojects {
+    repositories {
+        ...
+        maven { url 'https://jitpack.io' }
+    }
+}
+```
+
+Add the latest release appsDiscovery version to your *build.gradle* file
+
+
+```gradle
+dependencies {
+    ...
+    implementation 'com.github.kinecosystem.move-kin-android:appsDiscovery::<latest release>'
+}
+```
+
+For the latest release version go to [https://github.com/kinecosystem/move-kin-android/releases](https://github.com/kinecosystem/move-kin-android/releases).
+
+
+
+## Explore the Ecosystem
+
+- Adding a button that starts the experience of exploring the Ecosystem - button *AppsDiscoveryButton* 
+
+You can embed the button in any of your layout xml files
+click on the button starts the exploring experience 
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.constraint.ConstraintLayout>
+    ...
+    <org.kinecosystem.appsdiscovery.sender.discovery.view.customView.AppsDiscoveryButton
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            app:layout_constraintStart_toStartOf="parent" 
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintBottom_toBottomOf="parent"/>
```

Comment:
you could possibly leave out all the constraint layout part and just leave this:

```xml
<org.kinecosystem.appsdiscovery.sender.discovery.view.customView.AppsDiscoveryButton
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"/>
```